src/components/encounter/ohri-encounter-form.tsx

```diff
@@ -107,16 +92,8 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
     [encounter, encounterDate, form?.encounter, location, patient, previousEncounter, sessionMode],
   );
 
-  useEffect(() => {
-    if (!encounterLocation && location) {
-      setEncounterLocation(location);
-    }
-  }, [location]);
-
-  useEffect(() => {
-    const allFormFields: Array<OHRIFormField> = [];
-    const tempInitVals = {};
-    let isFieldEncounterBindingComplete = false;
+  const rawFormFields = useMemo(() => {
+    const ret = [];
```

Comment:
Temporarily holds the return value. This a [common naming pattern](https://www.reddit.com/r/learnprogramming/comments/31cep5/comment/cq0a475/?utm_source=share&utm_medium=web2x&context=3) and it's in short form because `return` is a reserved keyword

Reply:
Sorry for being old but I believe that would not be approved by uncle Bob because you never `return return;`. There is no meaning on that, `temp` or better `contextNameOfWhatYouAreReturning` is always better.  ;)

Reply:
Polished